packages/protocol/src/mdns/MdnsClient.ts

```diff
@@ -436,12 +446,20 @@ export class MdnsClient implements Scanner {
     #finishWaiter(queryId: string, resolvePromise: boolean, isUpdatedRecord = false) {
         const waiter = this.#recordWaiters.get(queryId);
         if (waiter === undefined) return;
-        const { timer, resolver, resolveOnUpdatedRecords, commissionable } = waiter;
+        const { timer, resolver, responder, resolveOnUpdatedRecords, commissionable } = waiter;
         if (isUpdatedRecord && !resolveOnUpdatedRecords) return;
         logger.debug(`Finishing waiter for query ${queryId}, resolving: ${resolvePromise}`);
         timer?.stop();
         if (resolvePromise) {
-            resolver();
+            const response = responder?.();
+            if (MaybePromise.is(response)) {
```

Comment:
Is there a code path where this returns a promise?  I tried to find it so I could make sure that something was taking the promise (which we don't do here) but it looks like the registered functions are always synchronous.

Reply:
The current usage is sync only ... I could also just do that but I thought still good to have for potential future things ... I can remove if you like it more "just sync for now"